source/extensions/transport_sockets/tls/context_impl.cc

```diff
@@ -336,6 +338,42 @@ ContextImpl::ContextImpl(Stats::Scope& scope, const Envoy::Ssl::ContextConfig& c
   //
   // Note that if a negotiated version is outside of this set, we'll issue an ENVOY_BUG.
   stat_name_set_->rememberBuiltins({"TLSv1", "TLSv1.1", "TLSv1.2", "TLSv1.3"});
+
+  if (!config.tlsKeyLogPath().empty()) {
+    ENVOY_LOG(debug, "Enable tls key log");
+    access_log_ = config.accessLogManager().createAccessLog(
+        Filesystem::FilePathAndType{Filesystem::DestinationType::File, config.tlsKeyLogPath()});
+    enableTlsKeyLog();
+  }
+}
+
+void ContextImpl::keylogCallback(const SSL* ssl, const char* line) {
+  ASSERT(ssl != nullptr);
```

Comment:
Why can we cast TransportSocketCallbacks to ContextImpl ? From the class definition,  TransportSocketCallbacks can only be cast to ConnectionImpl. Do you mean ConnectionImpl instead of ContextImpl ?
It is supposed that you mean ConnectionImpl, it can give use the connection info. But we still need the ContextImpl to get the configured local/remote filter info.

Reply:
Got it. I will fix.